AppleIntelWifiAdapter/AppleIntelWifiAdapterV2.cpp

```diff
@@ -401,13 +401,18 @@ IO80211Interface* AppleIntelWifiAdapterV2::getNetworkInterface() {
 }
 
 IOReturn AppleIntelWifiAdapterV2::getHardwareAddress(IOEthernetAddress *addrP) {
-    if(!this->drv->m_pDevice->ie_dev->address[0]) {
-        return kIOReturnError;
-    }
-    else {
-        memcpy(&addrP->bytes, &this->drv->m_pDevice->ie_dev->address, ETH_ALEN);
+    // Check if address is valid (all zeroes means invalid)
+    for (int i=0;i<ETH_ALEN;i++) {
+        if (this->drv->m_pDevice->ie_dev->address[i]) {
+            // Byte is non-zero, this address must be valid and we can return it
```

Comment:
Doesn’t this end up copying the entire hardware address every iteration?

Reply:
I don't think so? The `return` after the `memcpy()` should break the loop as well.

Reply:
My bad, I didn’t see the return.